### PR TITLE
[Performance] Optimize rotary embedding application in DeepSeek V4 by running in-place

### PR DESCRIPTION
**Pull request by ivyilike**
### What this PR does / why we need it?
`DSparkModel._project_shared_kv` now applies RoPE in place on the rope-segment view of the normalized kv tensor, instead of materializing the rope segment and re-assembling `[k_nope | k_pe]` with `split -> rope -> cat -> contiguous` . The previous implementation paid for a full extra read+write of the `[num_tokens, head_dim]` kv tensor on every call: the rope segment was detached via `split` , rotated into a fresh tensor, and then `torch.cat` reassembled a buffer that is numerically identical to `kv` itself (only the rope segment changes; the nope segment is passed through untouched). `ComplexExpRotaryEmbedding.forward` ends with `y.copy_(...)` , i.e. `npu_rotary_mul` writes its result back to the input storage. Running the rope directly on a view of `kv` therefore produces the same final values while the nope segment never moves and no concat buffer is allocated.

### Does this PR introduce _any_ user-facing change?
NO.

### How was this patch tested?
`pytest tests/ut/ops/test_rope_write_back.py -q`

- vLLM main: https://github.com/vllm-project/vllm/commit/84030bbe3d74d99bad477a3d2e37a973ccd8865c

### CODE REVIEW

**Comment by Tflowers-0129** [comment]
pls rebase, thx your contribution.

**Comment by ivyilike** [comment]
> pls rebase, thx your contribution.

okk

**Comment by ivyilike** [comment]
> pls rebase, thx your contribution.

 @Tflowers-0129  Rebased and CI is green ✅ Could you please approve when you get a chance? Thanks!

**Review by weijinqian0 — APPROVED** [review]
